src/unix/udp.c

```diff
@@ -657,15 +657,17 @@ int uv__udp_connect(uv_udp_t* handle,
 
 int uv__udp_disconnect(uv_udp_t* handle) {
     int r;
-    struct sockaddr addr;
 
-    memset(&addr, 0, sizeof(addr));
+    struct sockaddr_storage addr; 
+    socklen_t len = sizeof(addr);
+    getsockname(handle->io_watcher.fd, (struct sockaddr*)&addr, &len);
```

Comment:
Needs to check the return value of getsockname() and should be guarded with `#ifdef __APPLE__`.

As a rule of thumb: platforms where the existing code works okay shouldn't have to pay the price of an extra system call.

Style nit: can you add a `socklen_t len;` at the top of the function and assign it here? Libuv's style is mostly C89.

Reply:
will change it later